test/DSRAuthOracle.t.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "forge-std/Test.sol";
+
+import { DSRAuthOracle, IDSROracle } from "../src/DSRAuthOracle.sol";
+
+contract DSRAuthOracleTest is Test {
+
+    event SetPotData(IDSROracle.PotData nextData);
+
+    uint256 constant FIVE_PCT_APY_DSR        = 1.000000001547125957863212448e27;
+    uint256 constant ONE_HUNDRED_PCT_APY_DSR = 1.00000002197955315123915302e27;
+
+    DSRAuthOracle oracle;
+
+    function setUp() public {
+        // To get some reasonable timestamps that are not 1
+        skip(1 * (365 days));
+
+        oracle = new DSRAuthOracle();
+
+        // Feed initial data and set limits
+        oracle.grantRole(oracle.DATA_PROVIDER_ROLE(), address(this));
+        oracle.setMaxDSR(ONE_HUNDRED_PCT_APY_DSR);
+        oracle.setPotData(IDSROracle.PotData({
+            dsr: uint96(FIVE_PCT_APY_DSR),
+            chi: uint120(1e27),
+            rho: uint40(block.timestamp)
+        }));
+
+        skip(1 * (365 days));
+    }
+
+    function test_setPotData_rho_decreasing() public {
```

Comment:
Update all of these tests to be boundary tests, calling the function again with one unit higher or lower to demonstrate a corresponding success case

Reply:
fixed